### ISSUE TITLE
Upgrade Regex to 2023.8.8

### ISSUE THREAD

**Issue by epicserve** [squidfunk/mkdocs-material#6068]
### Description

In order to upgrade another project to mkdocs-material 9.4.1 I need mkdocs-material to upgrade regex to 2023.8.8. The following error happens when trying to upgrade to the newest version of mkdocs-material.

```
The conflict is caused by:
    The user requested regex==2023.8.8
    djlint 1.34.0 depends on regex<2024.0.0 and >=2023.0.0
    mkdocs-material 9.4.1 depends on regex~=2022.4
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

### Related links

- [Requesting a change](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/)
-

### Use Cases

See desccription.

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material/), related [issues](https://github.com/squidfunk/mkdocs-material/issues), and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by squidfunk** [comment]
Sure, we can do that. Would you like to submit a PR?

**Comment by squidfunk** [comment]
... I think the best way is if we switch regex back from the compatible release separator to a minimum version, because regex is a package that we use very sparingly (only in the search plugin), and we don't expect to break it, but other projects might be stuck on older or newer versions. Thus, we should switch to a minimum version for this dependency.

**Comment by epicserve** [comment]
Is the only change needed to change [this line](https://github.com/squidfunk/mkdocs-material/blob/7e6f15bb175f304547e379addf2f1daf4f351ede/requirements.txt#L33C1-L33C14) from `regex~=2022.4` to `regex>=2022.4`?

**Comment by squidfunk** [comment]
Yes.

**Comment by squidfunk** [comment]
Fixed in 43b5baf8a.

**Comment by squidfunk** [comment]
Released as part of [9.4.2](https://squidfunk.github.io/mkdocs-material/changelog/#9.4.2).
